### PR TITLE
fix: hide password change for ldap users

### PR DESCRIPTION
**Pull request by CuriouzK0d3r**
### Description of PR

hide password change page for ldap users

What component(s) does this PR affect?

  - [x] Back-End (Node.js, Next.js)
  - [x] Front-End (React.js)
  - [ ] Theming (CSS, React theme)
  - [ ] Build System
  - [ ] Docs
  
  
  
  Does the PR require changes on other components? If yes, please mark the components:
  
  - [ ] Back-End (Node.js, Next.js)
  - [ ] Front-End (React.js)
  - [ ] Theming (CSS, React theme)
  - [ ] Build System
  - [ ] Docs
  
  
  
### Related Issue
  
  
  
  
  
  
  Resolves #
  
### Type
  
  - [ ] Bug fix (non-breaking change which fixes an issue)
  - [ ] New feature (non-breaking change which adds functionality)
  - [ ] Breaking change (fix or feature that would cause existing functionality to change)
  - [ ] Docs update
  - [ ] None of the above
  
  
  
### Checklist:
  
  - [ ] I have read the **[contributing guide](https://github.com/FORTH-ICS-INSPIRE/artemis/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
  - [ ] This change requires a change in the documentation.
  - [ ] I have updated the documentation accordingly.
  >)`>>>)>

### CODE REVIEW

File: apps/artemis-web/utils/parsers.tsx [hunk 1]
```diff
@@ -5,7 +5,9 @@ import { formatDate, genTooltip } from './token';
 
 export function extractUser(req: any): any {
   if (!req.user) return null;
-
+  let type = 'user';
```
Comment (vkotronis) [inline apps/artemis-web/utils/parsers.tsx:8]: we also have a `extractLdapUser` function, shouldn't the type also be set there?

File: apps/artemis-web/pages/password_change.tsx [hunk 2]
```diff
@@ -29,8 +29,8 @@ const PasswordChangePage = (props) => {
   );
 };
 
-export default AuthHOC(PasswordChangePage, ['admin', 'user']);
+export default AuthHOC(PasswordChangePage, ['admin', 'user'], 'ldap-user');
```
Comment (vkotronis) [inline apps/artemis-web/pages/password_change.tsx:32]: why is the default 'ldap-user' and not 'user'?

File: apps/artemis-web/components/401-hoc/401-hoc.tsx [hunk 3]
```diff
@@ -20,7 +20,7 @@ const AuthHOC = (WrappedComponent, ACL = []) => {
       router.push('/login');
     }
 
-    if (user && !loading && !ACL.includes(user.role)) {
+    if (user && !loading && (!ACL.includes(user.role) || blockUserType === user.type)) {
```
Comment (vkotronis) [inline apps/artemis-web/components/401-hoc/401-hoc.tsx:23]: what does this do here exactly? What changes among ldap and normal user for the 401-hoc?

**Review by vkotronis — COMMENTED** [review]
In general LGTM, left minor comments and then can be merged

File: apps/artemis-web/utils/parsers.tsx (hunk 1 above)
Comment (CuriouzK0d3r) [reply apps/artemis-web/utils/parsers.tsx:8]: Nice thought, but extractLdapUser is only being used by the ldap login function, which is defacto about LDAP users :)

File: apps/artemis-web/pages/password_change.tsx (hunk 2 above)
Comment (CuriouzK0d3r) [reply apps/artemis-web/pages/password_change.tsx:32]: Its not a default value :) it just says block  the "ldap-user" !

File: apps/artemis-web/components/401-hoc/401-hoc.tsx (hunk 3 above)
Comment (CuriouzK0d3r) [reply apps/artemis-web/components/401-hoc/401-hoc.tsx:23]: It says if the user type is equal with the blockUserType, then block access.

The blockUserType is being passed by the specific page. e.g. if I want the system page to be unaccessible by ldap users, I can do that.

Comment (vkotronis) [reply apps/artemis-web/components/401-hoc/401-hoc.tsx:23]: I see, OK fine as-is then, good adjustment

File: apps/artemis-web/utils/parsers.tsx (hunk 1 above)
Comment (vkotronis) [reply apps/artemis-web/utils/parsers.tsx:8]: I get this, but shouldn't we also set the type there? Do we already set it? I.e., if someone connects over LDAP, the `type` var is set or not?

Comment (CuriouzK0d3r) [reply apps/artemis-web/utils/parsers.tsx:8]: Fair point, added that :)

Btw the var is not set in the database, only in the javascript object.

Comment (vkotronis) [reply apps/artemis-web/utils/parsers.tsx:8]: Correct, this is carried only on JS side

**Review by vkotronis — APPROVED** [review]
LGTM

**Pull request merged**